src/server/qgsserverparameters.cpp

```diff
@@ -200,6 +205,107 @@ QgsRectangle QgsServerParameterDefinition::toRectangle( bool &ok ) const
   return extent;
 }
 
+QByteArray QgsServerParameterDefinition::loadUrl( bool &ok ) const
+{
```

Comment:
Hi @nyalldawson,

I'm not very familiar with the `QgsNetworkAccessManager` class. Can you take a look on this method in particular to give me your opinion please?

Thanks :)

Reply:
I'm not familiar with the design of server, but calling  QCoreApplication::processEvents on the main thread is dangerous in app. I don't think that's an issue here as my understanding is that server does everything on the main thread?

There is a class QgsNetworkContentFetcher which handles the redirections etc for you, so it would be good to see that used to avoid duplicated code handling this.

Reply:
I can't use `QgsNetworkContentFetcher` because I can't use `connect` because `QgsServerParameterDefinition` does not derived QObject.

Reply:
Nyall is right, it would avoid a lot of duplicated code.

Why is it not possible to just use `QgsNetworkContentFetcher` in the same way as `QgsLayoutItemHtml` (for example in the `fetchHtml` method)?

Reply:
Has explain, `QgsServerParameterDefinition` is not derived from `QObject` so it's not possible to use `connect` in this context. So I can't do like `QgsLayoutItemHtml`
```
connect( mFetcher, &QgsNetworkContentFetcher::finished, &loop, [&loaded, &loop ] { loaded = true; loop.quit(); } );
```
If you have an idea, how I can use `QgsNetworkContentFetcher` I'm interested.

Reply:
Can't you call QObject:: connect?

Reply:
@pblottiere done ;-)

Reply:
Nice, it's much better this way. Thanks @rldhont :)